tools/fregrid/conserve_interp.c

```diff
@@ -92,304 +84,276 @@ void setup_conserve_interp(int ntiles_in, const Grid_config *grid_in, int ntiles
-	    interp[n].t_in   = (int    *)malloc(interp[n].nxgrid*sizeof(int   ));
-	    for(i=0; i<interp[n].nxgrid; i++) {
-	      interp[n].t_in [i] = m;
-	      interp[n].i_in [i] = i_in [i];
-	      interp[n].j_in [i] = j_in [i];
-	      interp[n].i_out[i] = i_out[i];
-	      interp[n].j_out[i] = j_out[i];
-	      interp[n].area[i]  = xgrid_area[i];
-	    }
-	    if(opcode & CONSERVE_ORDER2) {
-	      interp[n].di_in   = (double *)malloc(interp[n].nxgrid*sizeof(double));
-	      interp[n].dj_in   = (double *)malloc(interp[n].nxgrid*sizeof(double));
-	      for(i=0; i<interp[n].nxgrid; i++) {
-		interp[n].di_in [i] = xgrid_clon[i]/xgrid_area[i];
-		interp[n].dj_in [i] = xgrid_clat[i]/xgrid_area[i];
-	      }
-	    }
-	  }
-	  else {
-	    tmp_i_in  = interp[n].i_in;
-	    tmp_j_in  = interp[n].j_in;
-	    tmp_i_out = interp[n].i_out;
-	    tmp_j_out = interp[n].j_out;
-	    tmp_area  = interp[n].area;
-	    tmp_t_in  = interp[n].t_in;
-	    interp[n].i_in   = (int    *)malloc(interp[n].nxgrid*sizeof(int   ));
-	    interp[n].j_in   = (int    *)malloc(interp[n].nxgrid*sizeof(int   ));
-	    interp[n].i_out  = (int    *)malloc(interp[n].nxgrid*sizeof(int   ));
-	    interp[n].j_out  = (int    *)malloc(interp[n].nxgrid*sizeof(int   ));
-	    interp[n].area   = (double *)malloc(interp[n].nxgrid*sizeof(double));
-	    interp[n].t_in   = (int    *)malloc(interp[n].nxgrid*sizeof(int   ));
-	    for(i=0; i<nxgrid_prev; i++) {
-	      interp[n].t_in [i] = tmp_t_in [i];
-	      interp[n].i_in [i] = tmp_i_in [i];
-	      interp[n].j_in [i] = tmp_j_in [i];
-	      interp[n].i_out[i] = tmp_i_out[i];
-	      interp[n].j_out[i] = tmp_j_out[i];
-	      interp[n].area [i] = tmp_area [i];
-	    }
-	    for(i=0; i<nxgrid; i++) {
-	      ii = i + nxgrid_prev;
-	      interp[n].t_in [ii] = m;
-	      interp[n].i_in [ii] = i_in [i];
-	      interp[n].j_in [ii] = j_in [i];
-	      interp[n].i_out[ii] = i_out[i];
-	      interp[n].j_out[ii] = j_out[i];
-	      interp[n].area [ii] = xgrid_area[i];
-	    }
-	    if(opcode & CONSERVE_ORDER2) {
-	      tmp_di_in  = interp[n].di_in;
-	      tmp_dj_in  = interp[n].dj_in;
-	      interp[n].di_in   = (double *)malloc(interp[n].nxgrid*sizeof(double));
-	      interp[n].dj_in   = (double *)malloc(interp[n].nxgrid*sizeof(double));
-	      for(i=0; i<nxgrid_prev; i++) {
-		interp[n].di_in [i] = tmp_di_in [i];
-		interp[n].dj_in [i] = tmp_dj_in [i];
-	      }
-	      for(i=0; i<nxgrid; i++) {
-		ii = i + nxgrid_prev;
-		jj = j_in [i]*nx_in+i_in [i];
-		interp[n].di_in [ii] = xgrid_clon[i]/xgrid_area[i];
-		interp[n].dj_in [ii] = xgrid_clat[i]/xgrid_area[i];
-	      }
-	      free(tmp_di_in);
-	      free(tmp_dj_in);
-	    }
-	    free(tmp_t_in);
-	    free(tmp_i_in);
-	    free(tmp_j_in);
-	    free(tmp_i_out);
-	    free(tmp_j_out);
-	    free(tmp_area);
-	  }
-	}  /* if(nxgrid>0) */
-      }
-    }
+          }
+          else {
+            interp[n].i_in   = (int    *)realloc(interp[n].i_in,  interp[n].nxgrid*sizeof(int   ));
```

Comment:
I don't know why the spacing is so off.  Realloc is used here and all the tmp_* arrays have been removed